classes/Event/class.xoctEvent.php

```diff
@@ -631,11 +631,52 @@ public function getAnnotationLink() {
 			if (xoctConf::getConfig(xoctConf::F_SIGN_ANNOTATION_LINKS)) {
 				$this->annotation_url = xoctSecureLink::sign($url);
 			} else {
-
 				$this->annotation_url = $url;
 			}
 		}
 
+		// If Annnotation token security is enabled
+		if(xoctConf::getConfig(xoctConf::F_ANNOTATION_TOKEN_SEC))  {
+			// Get Media URL
+			$media_object= $this->getFirstPublicationMetadataForUsage(xoctPublicationUsage::find(xoctPublicationUsage::USAGE_ANNOTATE))->getMedia();	
+			$media_url = $media_object[0]->url;
+
+			if (xoctConf::getConfig(xoctConf::F_SIGN_PLAYER_LINKS)) {			
+				//get duration and calculate expire date
+				$duration = $media_object[0]->duration;
+				
+				$valid_until = null;
+				if (xoctConf::getConfig(xoctConf::F_SIGN_PLAYER_LINKS_OVERWRITE_DEFAULT)) {
+					$duration_in_seconds = $duration / 1000;
+					$additional_time_percent = xoctConf::getConfig(xoctConf::F_SIGN_PLAYER_LINKS_ADDITIONAL_TIME_PERCENT) / 100;
+					$valid_until = gmdate("Y-m-d\TH:i:s\Z", time() + $duration_in_seconds + $duration_in_seconds * $additional_time_percent);
+				}
+				// Sign the url and parse the variables.
+				$media_url_signed = xoctSecureLink::sign($media_url, $valid_until);
+				$media_url_query = parse_url($media_url_signed, PHP_URL_QUERY);
+				$media_url = $media_url . '&' . $media_url_query;
+			}
+
+			// Get user and course ref id
+			$ref_id = $_GET['ref_id'];
+			$user_id = xoctUser::getInstance(self::dic()->user())->getExtId();
```

Comment:
Is there a special reason you're using the Ext-ID and not the Identifier? The Identifier can be the Ext-ID or the email address, depending on the configuration.

Reply:
Yes! you have right, the reason to use the external ID is that it needs to be the same ID as used by opencast. This strategy will not be useful if Ilias and opencast use different IDs.